internal/app/machined/pkg/controllers/cluster/local_affiliate_test.go

```diff
@@ -35,6 +37,13 @@ func (suite *LocalAffiliateSuite) TestGeneration() {
 
 	suite.Require().NoError(suite.runtime.RegisterController(&clusterctrl.LocalAffiliateController{}))
 
+	mc := config.NewMachineConfig(container.NewV1Alpha1(&v1alpha1.Config{
+		ConfigVersion: "v1alpha1",
+		MachineConfig: &v1alpha1.MachineConfig{},
+		ClusterConfig: &v1alpha1.ClusterConfig{},
+	}))
+	suite.Require().NoError(suite.state.Create(suite.ctx, mc))
```

Comment:
we should assert on some control plane apiserver port in the test?

Reply:
No - if we go with optional machine config in previous suggestion, we can drop this block.

Reply:
I mean we should have a test that verifies that controlplane block gets properly populated :)